app/services/api/webflow-api.ts

```diff
@@ -31,76 +31,127 @@ import {
   cleanedWorkshops,
   convertScheduleToScheduleCard,
 } from "./webflow-helpers"
+import { queryClient } from "./react-query"
 
-const useWebflowAPI = <T>(key: string, collectionId: string, enabled = true) =>
-  useQuery({
-    queryKey: [key],
-    queryFn: async () => {
-      const { data } = await axiosInstance.get<PaginatedItems<T>>(
-        `/collections/${collectionId}/items`,
-      )
-      return data.items
-    },
-    enabled,
-  })
+const getCollectionById = async <T>(collectionId: string) => {
+  const { data } = await axiosInstance.get<PaginatedItems<T>>(`/collections/${collectionId}/items`)
+  return data.items
+}
+
+const useRecommendationsOptions = {
+  queryKey: [RECOMMENDATIONS.key, RECOMMENDATIONS.collectionId],
+  queryFn: async () => getCollectionById<RawRecommendations>(RECOMMENDATIONS.collectionId),
+} satisfies UseQueryOptions
```

Comment:
[q] since you're recreating this everywhere, wouldn't it be easier to make a recurring one?

for example:

```tsx
const useWebflowAPI = <T>(key: string, collectionId: string) =>
  useQuery({
    queryKey: [key, collectionId],
    queryFn: async () => {
      const { data } = await axiosInstance.get<PaginatedItems<T>>. 
        (`/collections/${collectionId}/items`)
      return data.items
    }
  })
```

Reply:
I'm pretty sure we need the `queryFn` reference to be same when we use it later in `useQueries` in `useScheduledEvents` and `prefetchQuery` in `prefetchScheduledEvents`. But I can check to see if that is true

Reply:
Refactored this to be less verbose in this commit: https://github.com/infinitered/ChainReactApp2023/pull/108/commits/1127ca1640d4b9f11fe6638fadb6b714e4d2799c